db/tables/operations/split.py

```diff
@@ -17,33 +19,24 @@ def _split_column_list(columns_, extracted_column_names):
     return extracted_columns, remainder_columns
 
 
-def _create_split_tables(extracted_table_name, extracted_columns, remainder_table_name, remainder_columns, schema, engine):
+def _create_split_tables(extracted_table_name, extracted_columns, remainder_table_name, schema, engine):
     extracted_table = create_mathesar_table(
         extracted_table_name,
         schema,
         extracted_columns,
         engine,
     )
-    remainder_fk_column = Column(
-        f"{extracted_table.name}_{constants.ID}",
-        ID_TYPE,
-        ForeignKey(f"{extracted_table.name}.{constants.ID}"),
-        nullable=False,
-    )
-    remainder_table = create_mathesar_table(
-        remainder_table_name,
-        schema,
-        [remainder_fk_column] + remainder_columns,
-        engine,
-        metadata=extracted_table.metadata
-    )
-    return extracted_table, remainder_table, remainder_fk_column.name
+    fk_column_name = f"{extracted_table.name}_{constants.ID}"
+    remainder_table_oid = get_oid_from_table(remainder_table_name, schema, engine)
+    extracted_table_oid = get_oid_from_table(extracted_table_name, schema, engine)
+    create_foreign_key_link(engine, schema, fk_column_name, remainder_table_oid, extracted_table_oid)
+    remainder_table_with_fk_key = reflect_table(remainder_table_name, schema, engine)
+    return extracted_table, remainder_table_with_fk_key, fk_column_name
 
 
-def _create_split_insert_stmt(old_table, extracted_table, extracted_columns, remainder_table, remainder_columns, remainder_fk_name):
+def _create_split_insert_stmt(old_table, extracted_table, extracted_columns, remainder_fk_name):
     SPLIT_ID = f"{constants.MATHESAR_PREFIX}_split_column_alias"
     extracted_column_names = [col.name for col in extracted_columns]
-    remainder_column_names = [col.name for col in remainder_columns]
     split_cte = select(
```

Comment:
We should create a bug ticket.The creation of this CTE will fail in cases where the extracted columns aren't orderable (e.g., JSON). I realize this bug is carried over from my own work.

Reply:
Created #1490 to track it

Reply:
Tracked by #1490